src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -702,11 +702,13 @@ public bool IsReadOnly
         public void Assign(string expression)
         {
             this.VerifyNotDisposed();
+            int threadId = Client.GetDebuggedThread().Id;
+            uint frameLevel = _ctx.Level;
```

Comment:
be careful here. Since this is all async, I think it is possible for the debuggee to get continued before this point. In which case, the frame could be gone or completely blow up.